src/Microsoft.VisualBasic/tests/StringsTests.cs

```diff
@@ -104,6 +104,99 @@ public void ChrW_CharCodeOutOfRange_ThrowsArgumentException(int charCode)
+        [InlineData(new string[] { "ba" }, "a", new string[] { "ba" }, new string[] { })]
+        [InlineData(new string[] { "bab" }, "a", new string[] { "bab" }, new string[] { })]
+        [InlineData(new string[] { "b" }, "a", new string[] { }, new string[] { "b" })]
+        [InlineData(new string[] { "a" }, "ab", new string[] { }, new string[] { "a" })]
+        [InlineData(new string[] { "ab" }, "ab", new string[] { "ab" }, new string[] { })]
+        public void Filter_SingleElement(string[] source, string match, string[] includeExpected, string[] excludeExpected)
+        {
+            Assert.Equal(includeExpected, Strings.Filter(source, match, Include: true));
+            Assert.Equal(excludeExpected, Strings.Filter(source, match, Include: false));
+        }
+
+        [Theory]
+        [InlineData(new string[] { "A" }, "a", new string[] { }, new string[] { "A" })]
+        public void Filter_SingleElement_BinaryCompare(string[] source, string match, string[] includeExpected, string[] excludeExpected)
+        {
+            Assert.Equal(includeExpected, Strings.Filter(source, match, Include: true, Compare: CompareMethod.Binary));
+            Assert.Equal(excludeExpected, Strings.Filter(source, match, Include: false, Compare: CompareMethod.Binary));
+        }
+
+        [Theory]
+        [InlineData(new string[] { "A" }, "a", new string[] { "A" }, new string[] { })]
+        public void Filter_SingleElement_TextCompare(string[] source, string match, string[] includeExpected, string[] excludeExpected)
+        {
+            Assert.Equal(includeExpected, Strings.Filter(source, match, Include: true, Compare: CompareMethod.Text));
+            Assert.Equal(excludeExpected, Strings.Filter(source, match, Include: false, Compare: CompareMethod.Text));
+        }
+
+        [Theory]
+        [InlineData(new string[] { "a", "a" }, "a", new string[] { "a", "a" }, new string[] { })]
+        [InlineData(new string[] { "a", "b" }, "a", new string[] { "a" }, new string[] { "b" })]
+        [InlineData(new string[] { "b", "a" }, "a", new string[] { "a" }, new string[] { "b" })]
+        [InlineData(new string[] { "b", "b" }, "a", new string[] { }, new string[] { "b", "b" })]
+        public void Filter_MultipleElements(string[] source, string match, string[] includeExpected, string[] excludeExpected)
+        {
+            Assert.Equal(includeExpected, Strings.Filter(source, match, Include: true));
+            Assert.Equal(excludeExpected, Strings.Filter(source, match, Include: false));
+        }
+
+        [Fact]
+        public void Filter_Objects_WhenObjectCannotBeConvertedToString_ThrowsArgumentOutOfRangeException()
+        {
+            var source = new object[] { typeof(object) };
+            var match = "a";
+
+            var exception = Assert.Throws<ArgumentException>(() => Strings.Filter(source, match));
+            var expectedMessage = "Argument 'Source' cannot be converted to type 'String'.\r\nParameter name: Source";
```

Comment:
We don't check exception strings in general. This test is failing because in some situations the line ending is different.  Instead on the AssertExtensions class are Throws overloads that at least verify the parameter name.